src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable jsx-a11y/click-events-have-key-events */
+/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
+/* eslint-disable quote-props */
+/* eslint-disable jsx-a11y/control-has-associated-label */
+import React, { useContext, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { TodoUpdateContext } from '../TodosProvider/TodosProvider';
+/* import cn from 'classnames'; */
+
+interface Props {
+  item: Todo;
+}
+
+export const TodoItem: React.FC<Props> = (props) => {
+  const { item: todo } = props;
+
+  const {
+    completeTodo,
+    uncompleteTodo,
+    editTodo,
+    deleteTodo,
+  } = useContext(TodoUpdateContext);
+
+  const [editing, setEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  let todoStatus = !todo.completed ? 'view' : 'completed';
```

Comment:
If you can write constant instead let use it prefer.
let worsens the readability and perception of the code during scaling

Reply:
next line: `if (editing) {
    todoStatus = 'editing';
  }`